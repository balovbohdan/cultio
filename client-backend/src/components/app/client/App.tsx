import * as React from 'react';

import {fetchHtml} from './utils';

type Props = {
    testRes?:any;
    testMode?:boolean;
};

export const App = (props:Props) => {
    const [__html, setHtml] = React.useState('');

    doFetchHtml(props)
        .then(({html}) => setHtml(html));

    return __html
        ? <div dangerouslySetInnerHTML={{ __html }}/>
        : null;
};

const doFetchHtml = async ({testMode, testRes}:Props) =>
    testMode
        ? testRes || { html: 'html' }
        : await fetchHtml();