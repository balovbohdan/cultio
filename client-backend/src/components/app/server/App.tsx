import * as React from 'react';
import '../../../css/index.css';

type Props = {
    devMode:boolean;
    children:React.ReactNode;
    state:{[key:string]:any};
};

export const App = ({state, devMode,children}:Props) =>
    <>
        <div id='app'>{children}</div>
        <HotReloadScript devMode={devMode}/>
        <ApolloState state={state}/>
    </>;

const HotReloadScript = ({devMode}) =>
    devMode
        ? <script src='reload/reload.js'/>
        : null;

const ApolloState = ({state}) => {
    const stateSerialized = JSON.stringify(state).replace(/</g, '\\u003c');
    const __html = `window.__APOLLO_STATE__ = ${stateSerialized};`;

    return <script dangerouslySetInnerHTML={{ __html }}/>;
};