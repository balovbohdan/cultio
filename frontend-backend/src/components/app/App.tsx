import '../../css/index.css';
import * as React from 'react';

type Props = {
    devMode:boolean;
    children:React.ReactNode;
    state:{[key:string]:any};
};

export const App = ({state, devMode,children}:Props) =>
    <body>
        <div id='app'>{children}</div>
        <HotReloadScript devMode={devMode}/>
        <ApolloState state={state}/>
    </body>;

const HotReloadScript = ({devMode}) =>
    devMode
        ? <script src='reload/reload.js'/>
        : null;

const ApolloState = ({state}) => {
    const stateSerialized = JSON.stringify(state).replace(/</g, '\\u003c');
    const __html = `window.__APOLLO_STATE__ = ${stateSerialized};`;

    return <script dangerouslySetInnerHTML={{ __html }}/>;
};