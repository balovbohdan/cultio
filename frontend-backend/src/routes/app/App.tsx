import * as React from 'react';
import {ApolloClient} from 'apollo-client';
import * as ReactDOMServer from 'react-dom/server';

import {Html} from './Html';
import {Content} from './Content';

export type Props = {
    location:string;
    devMode:boolean;
    client:ApolloClient<any>;
    context:{[key:string]:any};
};

export const App = (props:Props) => {
    const state = props.client.extract();
    const content = createContentStr(props);

    return <Html
        state={state}
        content={content}
        devMode={props.devMode}/>;
};

const createContentStr = ({context, client, location}:Props) => {
    const content = <Content
        client={client}
        context={context}
        location={location}/>;

    return ReactDOMServer.renderToString(content);
};