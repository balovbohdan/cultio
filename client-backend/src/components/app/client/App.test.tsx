import * as React from 'react';
import {shallow, mount} from 'enzyme';

import {act} from 'react-dom/test-utils';
import {render, unmountComponentAtNode} from 'react-dom';

import {App} from './App';

describe('app/client/App', () => {
    let container:null|HTMLDivElement = null;

    beforeEach(() => {
        container = document.createElement('div');

        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container && container.remove();
        container = null;
    });

    it('Should render correctly', () => {
        const app = shallow(<App testMode={true}/>);

        expect(app).toMatchSnapshot();
    });

    it('Should have HTML string in the state', async () => {
        await act(async () => {
            const testRes = { html: 'html' };
            const component = <App testMode={true} testRes={testRes}/>;

            render(component, container);
        });
    });
});