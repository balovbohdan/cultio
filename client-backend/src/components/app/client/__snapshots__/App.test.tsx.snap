// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/client/App Should render correctly 1`] = `""`;
